src/diagnostics.rs

```diff
@@ -588,6 +588,36 @@ pub fn report_msg<'tcx>(
         }
     }
 
+    // Traverse thread spawns to display thread backtraces up to main thread
+    if let Some(mut thread) = thread {
+        let mut thread_spawn_context = machine.threads.get_thread_spawn_context(thread);
+
+        while let Some((spawning_thread, thread_spawn_backtrace)) = thread_spawn_context {
+            err.note(format!(
+                "thread `{}` was spawned by thread `{}`",
+                machine.threads.get_thread_display_name(thread),
+                machine.threads.get_thread_display_name(*spawning_thread)
```

Comment:
Does this work if the thread has already finished?

Reply:
From the logic in join_thread and terminate_active_thread
https://github.com/rust-lang/miri/blob/21d531b37b51d80a997ef03be1fc09dfe40556b2/src/concurrency/thread.rs#L642
https://github.com/rust-lang/miri/blob/21d531b37b51d80a997ef03be1fc09dfe40556b2/src/concurrency/thread.rs#L985

I was under the impression that the thread struct itself never actually gets deallocated so that future joins can be checked to ensure they aren't repeat joins on the given thread, but I could be mistaken. Would it be better to be safe and save the whole backtrace with thread names on thread creation?